test/e2e/set/scyllacluster/scyllamanager.go

```diff
@@ -267,82 +272,431 @@ var _ = g.Describe("Scylla Manager integration", func() {
 		o.Expect(tasks.TaskListItemSlice).To(o.BeEmpty())
 	})
 
-	g.It("should register cluster and sync backup tasks", func() {
+	provideGCSCredentials := func(ctx context.Context, sc *scyllav1.ScyllaCluster) *scyllav1.ScyllaCluster {
+		scCopy := sc.DeepCopy()
+
+		gcServiceAccountKey := f.GetGCSServiceAccountKey()
+		o.Expect(gcServiceAccountKey).NotTo(o.BeEmpty())
+
+		secret := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "gcs-service-account-key-",
+			},
+			Data: map[string][]byte{
+				"gcs-service-account.json": gcServiceAccountKey,
+			},
+		}
+
+		secret, err := f.KubeClient().CoreV1().Secrets(f.Namespace()).Create(ctx, secret, metav1.CreateOptions{})
```

Comment:
this is setting up secrets, for other storage providers it might require additional actions, I wouldn't couple it with just getting the manifest

Reply:
Can do do something in the middle like a method that gives you the cluster definitions with secrets and you can make extra steps on your own, if needed?

 Should we use the workload identities?

Reply:
> Should we use the workload identities?

Iirc it was what I wanted to do at first when setting up object storage in CI, but it would either require knowing e2e namespaces in advance in the the CI script, or running google API from e2es, so it was breaking separation of concerns and we ended up with creating keys.

> Can do do something in the middle like a method that gives you the cluster definitions with secrets and you can make extra steps on your own, if needed?

I can, but why is that better than the current approach? I feel like that's premature given that it's the only test using it now, and may and up wrong when we have to support other storage provisioners, while the existing one is quite flexible.

Reply:
> Iirc it was what I wanted to do at first when setting up object storage in CI, but it would either require knowing e2e namespaces in advance in the the CI script, or running google API from e2es, so it was breaking separation of concerns and we ended up with creating keys.

makes sense

> I can, but why is that better than the current approach? I feel like that's premature given that it's the only test using it now, and may and up wrong when we have to support other storage provisioners, while the existing one is quite flexible.

we can do it later I guess